### PR TITLE
Feat: multiple project managers

### PR DESCRIPTION
**Pull request by dleard**
Allows adding multiple managers to a project.

`addManagerToRevision` and `deleteManagerFromRevision` mutations have a query in the mutation return. The record returned from `projectManagerFormChangesByLabel` is a composite return value that doesn't represent an actual single database record. This means that we can't use the connection directive to update the data. We may want to look into [subscriptions](https://relay.dev/docs/guided-tour/updating-data/graphql-subscriptions/) at some point. I think they may end up being necessary to handle concurrent editing.

We could also look into using [refetchable fragments](https://relay.dev/docs/guided-tour/refetching/refreshing-fragments/) to refresh the ProjectManagerForm fragment, though it kind of sounds like this is essentially the same as adding a query to the mutation return.

### CODE REVIEW

File: app/components/Form/ProjectManagerForm.tsx
Comment (matthieu-foucault) [inline app/components/Form/ProjectManagerForm.tsx:182]:
```suggestion
                    <label htmlFor={your_select_id}>{node.projectManagerLabel.label} (optional)</label>
```
In the current behaviour, all managers are optional, so this should be reflected in the label. 
We do have a `FieldLabel` component, which could be reused here. At least the `getRequiredLabel` function could be reused.

File: app/components/Form/ProjectManagerForm.tsx
Comment (matthieu-foucault) [inline app/components/Form/ProjectManagerForm.tsx:101]: can we have an optimistic response here? The component flashes with an empty value and then the value changes when the network request succeeds

File: app/components/Form/ProjectManagerForm.tsx
Comment (matthieu-foucault) [inline app/components/Form/ProjectManagerForm.tsx:211]: should be disabled if there is no manager selected, or if the mutation is already in flight for this manager. You may need to refactor the react fragment into a separate component to track individual mutations

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: app/next-env.d.ts
Comment (matthieu-foucault) [inline app/next-env.d.ts:2]: you need to update your local yarn packages. This was removed in the latest nextjs updage

File: app/components/Form/ProjectManagerForm.tsx
Comment (pbastia) [reply app/components/Form/ProjectManagerForm.tsx:211]: We could also make the clear button a noop if the mutation is in flight or if there is no manger selectd, I don't think that would harm the UX, things would be cleared again from the user's perspective

File: app/tests/unit/components/Form/ProjectManagerForm.test.tsx
Comment (pbastia) [inline app/tests/unit/components/Form/ProjectManagerForm.test.tsx:173]: We're missing a test that makes sure the `updateFormChangeMutation` is called when we update a dropdown field without clearing it

**Review by pbastia — CHANGES_REQUESTED** [review]
Nice! 👌  Just missing a small test

File: app/mutations/Manager/addManagerToRevision.ts
Comment (matthieu-foucault) [inline app/mutations/Manager/addManagerToRevision.ts:37]: https://bcgov-cas.promyze.app/#/CAS%20GGIRCS/home?practiceId=61e1eb156e4cd70fe286ec9f
You could reuse `ProjectManagerForm_query` here, or, if that fragment was split in two, `ProjectManagerForm_projectRevision`

File: app/components/Form/ProjectManagerForm.tsx
Comment (matthieu-foucault) [inline app/components/Form/ProjectManagerForm.tsx:108]: The action here needs to depend on the `form_change` `operation`:

- If `operation === "CREATE"`, then you can discard it, as the manager was added in the same revision
- If `operation === "UPDATE"`, then the `form_change.operation`  should be updated to `ARCHIVE`

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
Haven't done a pass on the code yet, but there's an issue with the screenshots:
image

"required" validation should not show if this is optional

File: app/components/Form/ProjectManagerForm.tsx
Comment (pbastia) [inline app/components/Form/ProjectManagerForm.tsx:211]: I feel this entire pattern of creating / updating / deleting form changes depending on whether we have a certain field or not will be reused over and over, just a noting a thought for later on how we could reuse that logic

Comment (matthieu-foucault) [reply app/components/Form/ProjectManagerForm.tsx:211]: Yes, this may be refactored into a hook. We can do that after we have two examples of this pattern (there will be a second one in #280 )

**Review by matthieu-foucault — APPROVED** [review]
-- nvm -- reviewing too many PRs in parallel

File: app/pages/cif/project-revision/[projectRevision]/index.tsx
Comment (pbastia) [inline app/pages/cif/project-revision/[projectRevision]/index.tsx:176]: I'd really like to find better wording for this function / callback if you have ideas, I'm quite unhappy with what I chose when I first implemented it.
This is just confusing

**Review by pbastia — COMMENTED** [review]
looks good! I'll let @matthieu-foucault approve if he has more comments

**Review by pbastia — APPROVED** [review]

File: app/data/jsonSchemaForm/projectManagerSchema.ts
Comment (matthieu-foucault) [inline app/data/jsonSchemaForm/projectManagerSchema.ts:5]: This is an interesting one, as this property is actually required in the database. Can we leave a comment below to explain why we don't want it to be required in the JSON schema?

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: app/components/Form/ProjectManagerForm.tsx
Comment (matthieu-foucault) [inline app/components/Form/ProjectManagerForm.tsx:226]: Don't need those two fragments

**Review by matthieu-foucault — APPROVED** [review]

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**
